source/slang/slang-check-decl.cpp

```diff
@@ -5340,15 +5394,23 @@ namespace Slang
         {
             lookupResult = lookUpMember(m_astBuilder, this, name, subType, nullptr, LookupMask::Default, LookupOptions::IgnoreBaseInterfaces);
 
-            if (!lookupResult.isValid())
-            {
-                // If we failed to look up a member with the name of the
-                // requirement, it may be possible that we can still synthesis the
-                // implementation if this is one of the known builtin requirements.
-                // Otherwise, report diagnostic now.
-                if (!requiredMemberDeclRef.getDecl()->hasModifier<BuiltinRequirementModifier>() &&
-                    !(requiredMemberDeclRef.as<GenericDecl>() &&
-                        getInner(requiredMemberDeclRef.as<GenericDecl>())->hasModifier<BuiltinRequirementModifier>()))
+            // If we failed to look up a member with the name of the
+            // requirement, it may be possible that we can still synthesis the
+            // implementation if this is one of the known builtin requirements.
+            // Otherwise, report diagnostic now.
+            if (!lookupResult.isValid() 
+                && !requiredMemberDeclRef.getDecl()->hasModifier<BuiltinRequirementModifier>()
```

Comment:
Should we make the default ctor requirement a BuiltinRequirement, like `IComparable::equals`?